pkg/httpd/server.go

```diff
@@ -0,0 +1,95 @@
+package httpd
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+var ShutdownTimeout = time.Second * 5
+
+type Server struct {
+	addr   string
+	lnAddr string
+	logger *zap.Logger
+	srv    *http.Server
+	done   chan struct{}
```

Comment:
done as used below can be a sync.WaitGroup.

Reply:
fair